searchlib/src/vespa/searchlib/common/bitvector.cpp

```diff
@@ -91,9 +91,10 @@ BitVector::clearInterval(Index start, Index end)
 }
 
 void
-BitVector::clearIntervalNoInvalidation(Index start, Index end)
+BitVector::clearIntervalNoInvalidation(Index start_in, Index end)
 {
-    if (start >= end) { return; }
+    Index start = std::max(start_in, getStartIndex());
+    if (start >= end || end == 0 || size() == 0) { return; }
```

Comment:
This can be simplified if end is set to std::min(end, size()) before checking for start >= end.
if end or size() is  0 then start >= end will evaluate to true, triggering the early return.

Reply:
Refactored range checking.